internal/runners/packages/list.go

```diff
@@ -72,12 +99,77 @@ func (l *List) Run(params ListRunParams, nstype model.NamespaceType) error {
 		return locale.WrapError(err, fmt.Sprintf("%s_err_cannot_fetch_checkpoint", nstype))
 	}
 
-	rows := newFilteredRequirementsTable(model.FilterCheckpointNamespace(checkpoint, model.NamespacePackage, model.NamespaceBundle), params.Name, nstype)
-	var plainOutput interface{} = rows
-	if len(rows) == 0 {
+	language, err := model.LanguageByCommit(*commit)
+	if err != nil {
+		return locale.WrapError(err, "err_package_list_language", "Unable to get language from project")
+	}
+	ns := ptr.To(model.NewNamespacePkgOrBundle(language.Name, nstype))
+
+	// Fetch resolved artifacts list for showing full version numbers, if possible.
+	var artifacts []*artifact.Artifact
+	if l.project != nil && params.Project == "" {
+		rt, err := runbitsRuntime.NewFromProject(l.project, target.TriggerPackage, l.analytics, l.svcModel, l.out, l.auth, l.cfg)
+		if err != nil {
+			return locale.WrapError(err, "err_package_list_runtime", "Could not initialize runtime")
+		}
+		artifacts, err = rt.ResolvedArtifacts()
+		if err != nil && !errs.Matches(err, store.ErrNoBuildPlanFile) {
+			return locale.WrapError(err, "err_package_list_artifacts", "Unable to resolve package versions")
+		}
+	}
+
+	requirements := model.FilterCheckpointNamespace(checkpoint, model.NamespacePackage, model.NamespaceBundle)
+	sort.SliceStable(requirements, func(i, j int) bool {
+		return strings.ToLower(requirements[i].Requirement) < strings.ToLower(requirements[j].Requirement)
+	})
+
+	requirementsPlainOutput := []requirementPlainOutput{}
+	requirementsOutput := []requirement{}
+
+	for _, req := range requirements {
+		if !strings.Contains(strings.ToLower(req.Requirement), strings.ToLower(params.Name)) {
+			continue
+		}
+
+		if !strings.HasPrefix(req.Namespace, nstype.Prefix()) {
+			continue
+		}
+
+		version := req.VersionConstraint
+		if version == "" {
+			version = locale.T("constraint_auto")
+		}
+
+		resolvedVersion := ""
+		for _, a := range artifacts {
+			if a.Namespace == ns.String() && a.Name == req.Requirement && version != *a.Version {
+				resolvedVersion = *a.Version
+				break
+			}
+		}
+
+		plainVersion := version
+		if resolvedVersion != "" {
+			plainVersion = locale.Tr("constraint_resolved", version, resolvedVersion)
+		}
+		requirementsPlainOutput = append(requirementsPlainOutput, requirementPlainOutput{
+			Name:    req.Requirement,
+			Version: plainVersion,
+		})
+
+		requirementsOutput = append(requirementsOutput, requirement{
+			Name:            req.Requirement,
+			Version:         version,
+			ResolvedVersion: resolvedVersion,
```

Comment:
Because we only set resolvedVersion if it does not equal the requested version it means that the structured output does not provide the `resolved_version` in this scenario. I think for structured output we should always set it, regardless of whether it matches the requested version. I think `state languages` will have the same issue.

Reply:
Okay, structured output now always gives the resolved version, both for `state languages` and `state packages`.